### DISCUSSION TITLE
Synchronize xml fields

### DISCUSSION THREAD

**Discussion by sojan1** [Mimetis/Dotmim.Sync#1096] [Q&A]
Hi 

i read through the post - #262

Is there any other way we can synchoronize xml fields between hub and device, as we currently have xml fields in few tables. 

I **tried using change tracking** it didn't go well either. Why would the Sync not work with change tracking? **if change tracking is based on rows** (not columns) **in the table that were changed after the last synchronisation?**
https://docs.microsoft.com/en-us/sql/relational-databases/track-changes/track-data-changes-sql-server?view=sql-server-ver15#Tracking

Thanks 
Sojan

**Comment by matevzcesar** [comment]
This was discussed in issue: #910 
You can try and use the solution form PR #941

**Reply by sojan1** [reply]
Hi thanks for responding.

 the solution #941 won't work for us as we are using an existing database, and we cannot change the schema either.

But i saw the suggession from @Mimetis  https://github.com/Mimetis/Dotmim.Sync/issues/910#issuecomment-1361658337

**Will XML column sync works if we add the xml field with a CAST to VARCHAR(MAX) in the stored procedure** and also add the field in BulkType?

![image](https://github.com/Mimetis/Dotmim.Sync/assets/111710684/9a93a6e3-bc1f-4a30-a6fe-a1f82c0a24fc)


![image](https://github.com/Mimetis/Dotmim.Sync/assets/111710684/94088e75-ba77-4693-95a7-5386ed82bb77)

Thanks
Sojan

**Reply by sojan1** [reply]
Hi @Mimetis 

is there anyway i can sync xml fields without converting the schema? Will it help if i add a **CAST to the stored procedure**? 
![image](https://github.com/Mimetis/Dotmim.Sync/assets/111710684/6bf355c9-bb6c-4f73-a142-c8a2f7584303)

**Reply by Mimetis** [reply]
I dont know.
Maybe @SvdSinner can help on that

**Reply by richard-sisson** [reply]
I work with Sojan and was wondering if there were any updates on this?

We've implemented a check in the code that creates the stored procedures, wrapping the field with the CAST code when the datatype is XML.  

This has enabled the provisioning of tables with XML fields and based on a small number of tests this appears to be working as expected.

Do you have any immediate concerns regarding approach?

Thanks
Richard
